### ISSUE TITLE
Several invalid escape sequences

### ISSUE THREAD

**Issue by wyattscarpenter** [UKPLab/sentence-transformers#2687]
As of python 3.12, there is a SyntaxWarning for strings that rely on the fact that `\` in front of a character with no present escape meaning results in a literal `\`. https://docs.python.org/3/whatsnew/3.12.html#other-language-changes

Seems like an easy fix. Either double the backslash or use a raw string, I suppose.

Here are the ones I came across, personally:
```python
.venv\Lib\site-packages\sentence_transformers\evaluation\SentenceEvaluator.py:81: SyntaxWarning: invalid escape sequence '\g'
  return re.sub(r"([a-z])([A-Z])", "\g<1> \g<2>", class_name)
.venv\Lib\site-packages\sentence_transformers\model_card.py:515: SyntaxWarning: invalid escape sequence '\d'
  if dataset_name and re.match("_dataset_\d+", dataset_name):
.venv\Lib\site-packages\sentence_transformers\losses\DenoisingAutoEncoderLoss.py:16: SyntaxWarning: invalid escape sequence '\_'
  """
  ```

**Comment by tomaarsen** [comment]
Hello!

Well spotted! I'll have a look at these in the coming days. I'll also be glad to accept a pull request on this in the meantime.

- Tom Aarsen

**Comment by tomaarsen** [comment]
Closed via #2765, thanks for this work!

- Tom Aarsen
